taichi/transforms/make_thread_local.cpp

```diff
@@ -86,8 +86,8 @@ std::vector<T *> find_global_reduction_destinations(
 }
 
 void make_thread_local_offload(OffloadedStmt *offload) {
-  // TODO: deal with struct for
-  if (offload->task_type != offload->range_for)
+  if (offload->task_type != OffloadedStmt::range_for &&
+      offload->task_type != OffloadedStmt::struct_for)
```

Comment:
This may make crash the Metal backend since IIRC TLS for struct-for is not yet supported there. We can either tolerate it for a while until we add struct-for TLS for Metal (if that's doable in a short amount of time), or we introduce a new `struct_for_tls` extension?

Reply:
I think it's fine to leave this as is. I expect that struct-for + TLS is used very rarely on non-LLVM backend, since they supports limited sparse data structures anyway... But in the meantime, I'll see what I can do to make this work for struct-for as well :)

Reply:
I see! I forgot that dense structures struct-fors are demoted to range-fors! Thanks!